lib/honeycomb-rails/config.rb

```diff
@@ -59,6 +60,10 @@ def record_flash?
     # instance during each request, and which should return a sample rate for
     # the request in question.
     attr_accessor :sample_rate
+
+    # If set to true, captures exception class name / message / backtrace
+    # source along with Rails request events. (default: false)
+    attr_accessor :capture_exceptions
```

Comment:
Why not default to `true`? It seems like the vast majority of use cases would prefer to know about their exceptions, and it's definitely something I think we'd promote as a best practice. Do we have a use case for someone actively preferring not to capture exceptions?

I can see the argument for disabling ActiveRecord instrumentation - it's runs in the happy path so it's expensive, and there are plenty of Rails apps that might use another database layer, not have a database at all, or just not care about their database calls. By contrast, exception capturing only happens when stuff is already broken, everyone cares about errors, and there's no alternative error-signalling approach someone could be using.

The only part of this functionality that I can imagine anyone wanting to turn off would be capturing the backtrace, to avoid sending large traces to Honeycomb. So I could see the case for a `config.capture_exception_backtraces` option much more clearly. (I've thought about adding a similar option to the ActiveRecord instrumentation too.)

Generally I think with a framework-level integration like this we have the opportunity to be a little more opinionated about instrumentation best practices than we can be in the SDK (which is general-purpose and lower-level). I'm all for configurability, but I think the choice of default behaviour is a great way to promote best practices, while still giving people the freedom to turn things off if they want. (That's also in line with the philosophy of Rails and its ecosystem.)

Reply:
Yeah, the exception class and message seem handy for nearly all cases, definitely like having those on by default. I do feel some need to tiptoe around how we discuss this though — the experience of using a good dedicated error monitoring product (Bugsnag, Sentry) is _so_ much better for errors right now (w/ breadcrumb trails, syntax highlight, repo/release cross-links, user-agent parsing, workflow states) that I really want to downplay the fact that we can do this. Forcing people to opt in to some error-related features helps enforce that framing. Fwiw I think what's on this PR right now is right, just... feel ⚠️  feelings whenever we wander into this territory.